pandas/core/internals/blocks.py

```diff
@@ -2234,12 +2237,16 @@ def new_block(
     *,
     ndim: int,
     refs: BlockValuesRefs | None = None,
+    dtype: DtypeObj | None,
 ) -> Block:
     # caller is responsible for ensuring:
     # - values is NOT a NumpyExtensionArray
     # - check_ndim/ensure_block_shape already checked
     # - maybe_coerce_values already called/unnecessary
-    klass = get_block_type(values.dtype)
+    if dtype:
+        klass = get_block_type(dtype)
+    else:
+        klass = get_block_type(values.dtype)
```

Comment:
as above, values.dtype should be the ListDtype already.  I don't see why passing dtype separately is necessary.